src/Lean/Syntax.lean

```diff
@@ -172,6 +172,24 @@ partial def getTailWithPos : Syntax → Option Syntax
   | node _ args         => args.findSomeRev? getTailWithPos
   | _                   => none
 
+open SourceInfo in
+/-- Splits an `ident` into its dot-separated components while preserving source info.
+For example, `` `foo.bla.boo `` ↦ `` [`foo, `bla, `boo] ``. -/
+def identComponents : Syntax → List Syntax
```

Comment:
I hoped to also use this in `Term.resolveName'` but there we also have to compute a substring like `foo.bla` (if `foo.bla` was a namespaced identifier), so doesn't quite fit.

Reply:
What about an optional parameter to keep the first `n` components as a single `Syntax`, then pass `n := id.components.length` from `resolveName'`, would that work?